### DISCUSSION TITLE
TrappingSR3 not working for pressure data set

### DISCUSSION THREAD

**Discussion by waleedaero** [dynamicslab/pysindy#308] [General]
Hello,

I optimised my trappingSR3 model based on the vorticity dataset (Oscillating cylinder). However, when I replace the data set with the pressure dataset (same problem, oscillating cylinder), the trappingSR3 fails to work. I have tried different threshold and eta values but cannot get it working. What can be the issue?

I'm using the same constraints. I just replaced the data set.

**Comment by waleedaero** [comment]
I'm also getting this error :  Solver failed, setting coefs to zeros

**Comment by akaptano** [comment]
I think you are misunderstanding how to use this optimizer -- take a look at the paper again (and use the trapping_extended branch of the code, which has a bug fix we have not yet merged into the main branch). The TrappingSR3 optimizer is used to build reduced-order models from velocity data (vorticity data should also work with some minor modification) of fluid flows. The pressure evolution and dataset cannot be used, since it will result in several important changes, e.g. the "energy" is now p^2 instead of u^2, and it is not guaranteed to be a Lyapunov function.

**Comment by waleedaero** [comment]
Hello, pardon me, but I didn't understand the trapping_extended part. I get an error: module 'pysindy' has no attribute 'trapping_extended' .
Secondly, overall SR3 performs very poorly for the pressure data, with the model not working for polynomial libraries above degree 2. 
I'm asking, because I think, I lack understanding.

**Comment by akaptano** [comment]
trapping_extended refers to a "branch" of the PySINDy repository (https://github.com/dynamicslab/pysindy/tree/trapping_extended). This is a version of PySINDy with some different functionality. 

Does the model fit the X_dot data well or do you find errors when you simulate? This issue with stability is a common problem.

**Reply by waleedaero** [reply]
I have a high relative error of 0.9 and a model score of 0.5 for SR3.
But it works great with STLSQ

**Reply by waleedaero** [reply]
the relative error is reconstruction error, reconstructing the training data.

**Comment by waleedaero** [comment]
I have a question, that has been bothering me for a while and for which I haven't been able to find a satisfactory answer. Why does the vorticity data perform better than the pressure dataset? In general, I have seen this for STLSQ and SR3 optimizers as well.
